### PR TITLE
Updated keycloak.py to allow defining connection timeout value (#4168)

### PR DESCRIPTION
**Pull request by laskasn**
##### SUMMARY

You can find a summary along with a discussion [here](https://github.com/ansible-collections/community.general/issues/4168) 

**IMPORTANT:** This pull request adds a new parameter to the keycloak module, the `connection_timeout`. 
If provided, it propagates this value as the optional value `timeout` to the `open_url()` calls. By default (if not defined), it inherits the 10 sec default value of the underlying `timeout` parameter. 
We decided to name it `connection_timeout` instead of just `timeout`, because we didn't want to have a naming collision with the task timeout parameter or the network-specific ones, as mentioned [here](https://docs.ansible.com/ansible/latest/network/user_guide/network_debug_troubleshooting.html#command-timeout), but we could rename it into anything else in case this name is not appropriate.
 
The [documentation](https://docs.ansible.com/ansible/latest/collections/community/general/index.html) should also be updated, to include this parameter as an optional to all keycloak plugin calls.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
[keycloak](https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/identity/keycloak/keycloak.py)

### CODE REVIEW

File: changelogs/fragments/4168-add-keycloak-url-timeout.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "Keycloak module - added connection timeout parameter when calling server (https://github.com/ansible-collections/community.general/pull/4168)."
```
Comment (felixfontein) [inline changelogs/fragments/4168-add-keycloak-url-timeout.yml:2]:
```suggestion
  - "keycloak_* modules - added connection timeout parameter when calling server (https://github.com/ansible-collections/community.general/pull/4168)."
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

You also need to document the new parameter in plugins/doc_fragments/keycloak.py.

File: changelogs/fragments/4168-add-keycloak-url-timeout.yml (hunk 1 above)
Comment (laskasn) [reply changelogs/fragments/4168-add-keycloak-url-timeout.yml:2]: Oh, ok, that's why it was complaining! Thanks a lot!

File: plugins/doc_fragments/keycloak.py [hunk 2]
```diff
@@ -61,4 +61,10 @@ class ModuleDocFragment(object):
             - Verify TLS certificates (do not disable this in production).
         type: bool
         default: yes
+    
+    connection_timeout:
+        description:
+            - Controls the http connections timeout period (in seconds) to keycloak API.
+        type: int
+        default: 10
```
Comment (felixfontein) [inline plugins/doc_fragments/keycloak.py:69]:
```suggestion
        default: 10
        version_added: 4.5.0
```

File: plugins/doc_fragments/keycloak.py [hunk 3]
```diff
@@ -61,4 +61,10 @@ class ModuleDocFragment(object):
             - Verify TLS certificates (do not disable this in production).
         type: bool
         default: yes
+    
+    connection_timeout:
+        description:
+            - Controls the http connections timeout period (in seconds) to keycloak API.
```
Comment (felixfontein) [inline plugins/doc_fragments/keycloak.py:67]:
```suggestion
            - Controls the HTTP connections timeout period (in seconds) to Keycloak API.
```

**Review by felixfontein — COMMENTED** [review]
Please check out the linting errors reported by CI, these also need to be fixed before merging.

**Comment by felixfontein** [comment]
CC @kris2kris @pmdumuid @laurpaum

**Comment by felixfontein** [comment]
Also CC @fynncfchen since this will affect the new modules PRs. The code they add (to module_utils) will have to be modified to use the timeout parameter as well.

**Review by kris2kris — APPROVED** [review]

**Review by pmdumuid — APPROVED** [review]

**Comment by felixfontein** [comment]
@laskasn thanks for contributing this!
@kris2kris @pmdumuid thanks for reviewing!

**Pull request merged**